tests: add filter for some tests using newlib

IRISZZW opened the pull request:
some tests configured with `CONFIG_NEWLIB_LIBC=y`, 
it's better to add a filter `filter: TOOLCHAIN_HAS_NEWLIB == 1`
in those tests yaml file.

pabigot reviewed (changes requested):
For completeness please look for other cases where this is necessary.  It appears that would include:
* tests/lib/gui/lvgl/testcase.yaml
* samples/net/syslog_net/sample.yaml

ruuddw commented:
@IRISZWW, can you have a look at other tests with newlib dependences like Peter suggested? Then we can get this one merged.

evgeniy-paltsev commented:
@IRISZWW, could you please look at this, so we can get this one merged.
Thanks.

pabigot commented on tests/application_development/libcxx/testcase.yaml:
```diff
@@ -2,11 +2,13 @@ tests:
   application_development.cpp.libcxx:
     platform_exclude: qemu_x86_coverage
     toolchain_exclude: xcc
+    filter: TOOLCHAIN_HAS_NEWLIB == 1
```
Just as a note: technically this isn't right, since what we really require is a libcxx, whether that comes from newlib or not.  Ultimately this should be `TOOLCHAIN_HAS_LIBCXX == ` but for now this is OK.

pabigot commented:
> For completeness please look for other cases where this is necessary. It appears that would include:

`git grep CONFIG_NEWLIB_LIBC=y  tests/` shows around 30 tests that have `CONFIG_NEWLIB_LIBC=y`, and 24 in `samples/`.  Only some of each have been updated to filter on the toolchain operation.

IRISZZW commented:
buildkite failures caused by ```undefined reference to `mpsl_fem_abort_extend'```,  not by this PR.

pabigot reviewed (changes requested):
Thanks for doing this.  I think this one still needs the filter:

* samples/net/sockets/tcp selects newlib in `prj.conf`

but AFAICT that's the only one missing.

pabigot commented:
@nashif is the change in this PR desired?  It makes sense, but....

rlubos commented:
> buildkite failures caused by `` undefined reference to `mpsl_fem_abort_extend' ``, not by this PR.

@IRISZZW This was caused by the nRF 802154 radio driver, the fix was just merged, please rebase.

IRISZZW commented:
> Thanks for doing this. I think this one still needs the filter:
> 
> * samples/net/sockets/tcp selects newlib in `prj.conf`
> 
> but AFAICT that's the only one missing.

there is no test in yaml file, do we really need to add the filter? 
https://github.com/zephyrproject-rtos/zephyr/blob/87ddddae5222f3158f3d8d4dec26867fc4ab4b4b/samples/net/sockets/tcp/sample.yaml#L1-L3

pabigot commented:
> > * samples/net/sockets/tcp selects newlib in `prj.conf`
> 
> there is no test in yaml file, do we really need to add the filter?

Nope, that's fine then.

pabigot commented:
@IRISZZW You can fix the compliance failure by adding something to the body of the commit, e.g. an explanation for why this is the right thing to do, like in the description for the PR.

IRISZZW committed:
tests: add filter for some tests using newlib

some tests configured with CONFIG_NEWLIB_LIBC=y,
it's better to add a filter filter: TOOLCHAIN_HAS_NEWLIB == 1
in those tests yaml file.

Signed-off-by: Watson Zeng <zhiwei@synopsys.com>
